### PR TITLE
Add functionality: `count_ones` and `count_zeros`

### PR DESCRIPTION
**Pull request by BenjaminRemez**

### CODE REVIEW

File: src/Modulo2.jl [hunk 1]
```diff
@@ -663,6 +679,16 @@ function copyto!(a::ZZ2Array{N}, bc::Broadcasted{ZZ2ArrayStyle{N}, <:Any, typeof
     a
 end
 
+#
+# Utility functions
+#
+
+count_ones(a::ZZ2) = ifelse(a.m, 1, 0)
+count_zero(a::ZZ2) = ifelse(a.m, 0, 1)
```
Comment (matthias314) [inline src/Modulo2.jl]:
```suggestion
count_ones(a::ZZ2) = Int(isodd(a))
count_zeros(a::ZZ2) = Int(iseven(a))
```
This way we use existing functions.

File: src/Modulo2.jl [hunk 2]
```diff
@@ -663,6 +679,16 @@ function copyto!(a::ZZ2Array{N}, bc::Broadcasted{ZZ2ArrayStyle{N}, <:Any, typeof
...
+# TODO: use views to not reduce over the padding to multiples of 256?
+count_ones(arr::ZZ2Array) = mapreduce(count_ones, +, arr.data; init = 0)
```
Comment (matthias314) [inline src/Modulo2.jl]:
```suggestion
count_ones(arr::ZZ2Array) = sum(count_ones, arr.data; init = 0)
```
This is more idiomatic and as fast as boiling it down to `mapreduce`.

**Review by matthias314 — APPROVED** [review]

**Comment by matthias314** [comment]
Thanks for these ideas!

It seems to me that the analogous functions for `BitArray` and `BitVector` don't exist in Julia. That makes me a bit hesitant. What use cases do you have in mind?

**Comment by BenjaminRemez** [comment]
I was actually unpleasantly surprised that BitVector does not have this functionality, and did not want to type-pirate it. The concrete use case I have (thank you for providing me this package!) is to get the Hamming weight of a bitstring.

**Pull request closed without merging**